src/main/java/org/terasology/module/health/systems/BlockDamageAuthoritySystem.java

```diff
@@ -233,21 +237,39 @@ public void onAttackHealthlessActAsBlock(AttackEvent event, EntityRef targetEnti
         }
     }
 
-    /** Adds health component to blocks when damaged. */
+    /**
+     * Adds health component to blocks when damaged.
+     */
     @ReceiveEvent
-    public void beforeDamagedEnsureHealthPresent(BeforeDamagedEvent event, EntityRef blockEntity, BlockComponent blockComponent) {
-        if (!blockEntity.hasComponent(HealthComponent.class)) {
-            Block type = blockComponent.getBlock();
-            if (type.isDestructible()) {
+    public void beforeDamagedEnsureHealthPresent(BeforeDamagedEvent event, EntityRef blockEntity,
+                                                 BlockComponent blockComponent) {
+        Block type = blockComponent.getBlock();
+        if (type.isDestructible()) {
+
+            if (!blockEntity.hasComponent(HealthComponent.class)) {
                 HealthComponent healthComponent = new HealthComponent();
                 healthComponent.maxHealth = type.getHardness();
                 healthComponent.currentHealth = type.getHardness();
                 healthComponent.destroyEntityOnNoHealth = true;
-                healthComponent.regenRate = type.getHardness() / BLOCK_REGEN_SECONDS;
-                healthComponent.waitBeforeRegen = 1f;
 
                 blockEntity.addComponent(healthComponent);
             }
+
+            if (!blockEntity.hasComponent(BaseRegenComponent.class)) {
+                BaseRegenComponent baseRegenComponent = new BaseRegenComponent();
+                baseRegenComponent.regenRate = type.getHardness() / BLOCK_REGEN_SECONDS;
+                baseRegenComponent.waitBeforeRegen = 1f;
+
+                blockEntity.addComponent(baseRegenComponent);
+            }
+
+            if (!blockEntity.hasComponent(RegenComponent.class)) {
+                //TODO: should this just send a registration event instead of creating the component on it's own?
```

Comment:
if we want to treat base regen like any other regen, then I'd say yes

Reply:
yep, I'll leave this to future me when reworking the damage part or the block handling part.